### PR TITLE
Sanitise Configuration State for OpenGL on macOS

### PR DESCRIPTION
**Pull request by IsaacMarovitz**
+ Check when loading the configuration file if the graphics backend has been set to OpenGL and force switch to Vulkan
+ Change default graphics backend on macOS to Vulkan
+ If graphics backend is changed to OpenGL at runtime, force back to Vulkan

### CODE REVIEW

File: Ryujinx.Ui.Common/Configuration/ConfigurationState.cs [hunk 1]
```diff
@@ -1191,6 +1191,13 @@ public ConfigurationLoadResult Load(ConfigurationFileFormat configurationFileFor
 
                 configurationFileFormat.EnableMacroHLE = true;
             }
+            
+            if (OperatingSystem.IsMacOS() && configurationFileFormat.GraphicsBackend == GraphicsBackend.OpenGl)
```
Comment (AcK77) [inline Ryujinx.Ui.Common/Configuration/ConfigurationState.cs:1195]: This should be checked In `LogValueChange` by checking the type and then sets the right value if it’s macOS.

You could add a `using` for the Logging class too.

**Comment by AcK77** [comment]
Instead of put a logging, adding a check in LoadDefault and when the backend is changed, it could be probably better to force the GraphicBackend field to Vulkan everytime if it's on macOS ?

**Comment by gdkchan** [comment]
I think it would be better if we had a way to check if each backend is supported on the current platform. Then we could take action from here, I think showing an error message saying that OpenGL or Vulkan is not supported would be enough.

**Comment by IsaacMarovitz** [comment]
> I think it would be better if we had a way to check if each backend is supported on the current platform. Then we could take action from here, I think showing an error message saying that OpenGL or Vulkan is not supported would be enough.

I will generalise the approach in this PR slightly in order to make it more customisable in future.

**Pull request closed without merging**